content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,273 @@
+you do not need to write the instructions on what exact steps need to be taken.
+Instead, you describe what your infrastructure should ultimately look like.
+Behind the scenes, Terraform will figure out what needs to be done. If you've
+cloned the repo, you'll find the `main.tf`file in the root of the project --
+that's where we will be configuring Terraform.
+
+## Terraform Provider Iterative
+
+Terraform can orchestrate a plethora of various resources for you, but for the
+majority of projects you only need a few. That's why instead of shipping all the
+integrations in one bundle, you install a barebones Terraform distribution and
+then plug in support for whatever resources you need with so-called
+[_providers_](https://www.terraform.io/docs/extend/how-terraform-works.html).
+For this tutorial we will only need
+[Iterative Provider](https://registry.terraform.io/providers/iterative/iterative/latest).
+It enables full lifecycle management of computing resources for machine learning
+pipelines from AWS, Microsoft Azure, Google Cloud Platform, and more. The
+Iterative Provider has a couple of advantages for machine learning pipelines.
+Namely,
+
+- The configuration for various cloud compute providers with the Iterative
+  Provider will be nearly identical, so you can easily migrate from one cloud
+  provider to another, if you want to.
+- It is designed not only to provision infrastructure but to execute your
+  scripts on it too, all via a single configuration.
+- It helps to sync data between your local machine and a remote one.
+- Once your training is complete, the remote resources will be terminated,
+  avoiding unused machines quietly ramping up costs.
+
+To start using the Iterative Provider we need to let Terraform know about it by
+adding the following in our `main.tf`:
+
+```hcl
+terraform {
+    required_providers { iterative = { source = "iterative/iterative", version = ">= 0.9.9" } }
+}
+provider "iterative" {}
+```
+
+Once you describe what providers you'll be using in your `main.tf`, run the
+`terraform init` command. If you have cloned the example repo, you should run
+this command before doing anything else. This will initialize your working
+directory and download the required provider.
+
+## Configuring `iterative_task`
+
+The Iterative Provider offers a single resource called `iterative_task` that
+we'll need to configure. This resource will:
+
+1. Create cloud resources (storage, machines) for the task.
+2. If specified, upload a directory to the cloud storage.
+3. Run the given script until completion or timeout in the cloud.
+4. Download results.
+5. Automatically terminate compute resources upon task completion.
+
+This is exactly what we need to run a model training process! Let's see how we
+can configure the `iterative_task`. First, I'll show you what my configuration
+file looks like, and then we'll unpack what's going on here:
+
+```hcl
+terraform {
```

Comment:
Maybe clarify that the code block below is the same main.tf file but edited to add the task.